big-red-shuttle/StopsViewController.swift

```diff
@@ -60,8 +62,43 @@ class StopsViewController: UIViewController, UITableViewDelegate, UITableViewDat
         let locations = stops.map { $0.getCoordinate() }
         setLocations(locations: locations)
         viewIsSetup = true
+        
+        //observe gps data changes
+        system().gps().addObserver(observer: self)
+        Location.shared.fetchedUserLocationCompletionBlock = {
+            if !system().gps().registeredToLogLocation {
+                system().api().registerUserToLogLocation(key: "fedorko", success: { (json: JSON?) in
+                    system().gps().startLoggingShuttleLocation()
+                    system().gps().startFetchingShuttleLocation()
+                }, failure: nil)
+            }
+        }
+    }
+    
+    //MARK: - Shuttle Bus GPS Delegate Protocol Methods
+    func gps(gps: GPS, movedToCoordinate coordinate: Coordinate) {
```

Comment:
I'm not clear on the purpose of GPS. Currently, you have shared access (should you?) yet here you are passing it around as a parameter. What is the point of this? It would be the equivalent of passing around `UserDefaults.standard` or `FileManager.default`. I think a tougher team review or discussion should be in place for the overall architecture.